src/test/suite/parser/parser.test.ts

```diff
@@ -36,6 +36,71 @@ suite('Parser', () => {
         });
     });
 
+    suite('getHotkeyByLine', () => {
+        // List of test data
+        let dataList = [
+            // {
+            //     in: // input test string
+            //     hn: // hotkey's name
+            // },
+            {
+                in: 'F1:: Goto ::btw', // GoTo HotString ::btw::
+                hn: 'F1',
+            },
+            {
+                in: 'F1:: Goto :*:btw', // GoTo HotString :*:btw::
+                hn: 'F1',
+            },
+            {
+                in: 'Hotkey If True',
+                hn: '',
+            },
+            {
+                in: 'Hotkey, If True',
+                hn: '',
+            },
+            {
+                in: 'Hotkey IfWinActive, Untitled',
+                hn: '',
+            },
+            {
+                in: 'Hotkey, IfWinActive, Untitled',
+                hn: '',
+            },
+            {
+                in: 'Hotkey F1',
+                hn: 'F1',
+            },
+            {
+                in: 'Hotkey, F1',
+                hn: 'F1',
+            },
+            {
+                in: 'Hotkey F1, Label',
+                hn: 'F1',
+            },
+            {
+                in: 'Hotkey, F1, Label',
+                hn: 'F1',
+            },
+        ];
+        dataList.forEach((data) => {
+            test("'" + data.in + "' => '" + data.hn + "'", async () => {
+                const document = await vscode.workspace.openTextDocument({
+                    language: 'ahk',
+                    content: data.in,
+                });
+                // Use array access for the private members
```

Comment:
Again, let's try to separate logic into internal and external and so we don't have to worry about hacky access of private members. We can just test the internal logic (the pure helper functions) for now since we didn't have tests for the external logic (connections to the VS Code API) before anyway.

Reply:
I tried, but it only made things worse.

I divided the function `Parser.getHotkeyByLine` into two parts: the first part had to be returned to `buildScript()` (it created an object of type `Hotkey`), the second part went into the internal logic (purely `RegExp` function).

As for me, this turns out even worse than it was.
Just put the entire RegExps into separate functions and test them? Just for the sake of not using a hack with access to private methods? Can the methods be made public?

Reply:
It's been a while :) go ahead and put things in different places, or make stuff public. I'll review when you think the changes are ready, feel free to @ me. Thanks again for all your contributions :)